### PR TITLE
Update product-technical-writing-team-handbook.md

### PR DESCRIPTION
**Pull request by justinegeffen**
This PR includes new workflow documentation as well as changes for the handbook (which are now outdated). The handbook section needs to be revised in terms of the verticals, DRIs, and allocation of AORs. This will be done in the next PR. 

ETA: I reviewed those sections and they seem up to date for now. @wiersgallak, @jasonblais: are those DRIs for the PR reviews still accurate for now?

### CODE REVIEW

**Review by cwarnermm — DISMISSED** [review]
Fantastic updates!

File: operations/research-and-development/product/technical-writing/product-technical-writing-team-handbook.md [hunk 1]
```diff
@@ -4,112 +4,114 @@ description: 1% Draft
...
+  * [Administrator's Guide](https://docs.mattermost.com/guides/administrator.html)
+  * Mattermost Cloud
+  * Apps Framework and Marketplace
+  * Self-managed deployments
+  * Messaging features and functionality
+* Incident Collaboration
+* Focalboard
+* Growth
+  * Customer Portal
+  * Compliance
```
Comment (jasonblais) [inline operations/research-and-development/product/technical-writing/product-technical-writing-team-handbook.md]: Let's replace Compliance with `Telemetry`

**Review by jasonblais — DISMISSED** [review]
Just one minor note, thanks Justine!

**Comment by jasonblais** [comment]
(I didn't review the technical writing workflows step-by-step, but at high-level they make sense)

File: operations/research-and-development/product/technical-writing/product-technical-writing-team-handbook.md (hunk 1 above)
Comment (justinegeffen) [inline operations/research-and-development/product/technical-writing/product-technical-writing-team-handbook.md]:
```suggestion
  * Telemetry
```

**Review by jasonblais — APPROVED** [review]

**Review by cwarnermm — APPROVED** [review]
Thanks!

**Review by wiersgallak — APPROVED** [review]

**Pull request merged**